client/v3/client.go

```diff
@@ -297,16 +297,23 @@ func (c *Client) dial(creds grpccredentials.TransportCredentials, dopts ...grpc.
 		dctx, cancel = context.WithTimeout(c.ctx, c.cfg.DialTimeout)
 		defer cancel() // TODO: Is this right for cases where grpc.WithBlock() is not set on the dial options?
 	}
-
-	initialEndpoints := strings.Join(c.Endpoints(), ";")
-	target := fmt.Sprintf("%s://%p/#initially=[%s]", resolver.Schema, c, initialEndpoints)
+	// Using "scheme://endpoint/authority" format. Endpoint passed doesn't matter as we are using custom resolver.
+	target := fmt.Sprintf("%s://%p/%s", resolver.Schema, c, authority(c.endpoints[0]))
```

Comment:
I thought we keep the overriden target... just add: 

```
opts := opts.WithAuthority(authority(c.endpoints[0]))
```

to ensure Authority is syntactically correct.

Reply:
Done

Reply:
Why it's better to craft URL that contains authority in the new format over the WithAuthority  option ?

I'm not opposing this change, but would like to learn.

Reply:
WithAuthority doesn't work with in TLS config.

Reply:
https://github.com/grpc/grpc-go/blob/1c598a11a4e503e1cfd500999c040e72072dc16b/dialoptions.go#L484-L491

Reply:
Thank you. Please add a comment pointing on this issue in grpc: https://github.com/grpc/grpc-go/issues/4717 for posterity.

Reply:
Done